### DISCUSSION TITLE
Feedback on the Android version

### DISCUSSION THREAD

**Discussion by ClementBeal** [ubuntu-flutter-community/musicpod#1063] [Ideas]
Hi!

I've tested your app on the Android version. I have only tested the "music" part because it's the only think I use. 

In general, it's good! There's some pages that I think are great! The "3 vertical dots" page for the tracks is really pretty. So now the few things that, according to me, could be improved!

- I don't think the "back arrow" in the AppBar is useful. I already have a back button that does the same thing and it's easier to reach.

I think it's ok to keeep the back arrow on a secondary page like the album page. On the main pages of the bottom bar navigation, we should remove it

- The special view "listening song and I see the album cover with the player control", I would be more ergonomic  if we can close this page with the Android's back button. Or even a slide down could be nice. Like with a bottomSheetModal.

- [x] I don't know if it's useful to have a "volume slide". I prefer to use the volume buttons of the phone.

- After ~10/15 navigations between random pages, the UI starts to be laggy. It could be wise to use `replace` instead of `push`. I suspect the "Album Grid page" to be heavy. It has to keep in memory dozens of images that are more or less heavy.
- When I press the "share button" from the "current listening track page", it shows nothing. I have to quickly close the page to see the white modal

- The favorites should be easy to access. It should be the first one. 
- The "artists" page : if there's no album cover, the artist's name is not clipped.
- Maybe some Hero animation could be nice. For example, if you press the album cover from the listening bar, the image move to the new position.
- I don't know if a click on the artist or track name in the listening bar should redirect to the artist/album page. It's hard to click with precision with my big fingers on the artist's name.
- If the track has no album cover, it shows an icon/widget different than the one from the Album page. It's black isntead of being green.
- The audio controller that are accessible in the Android panel (idk how is it called but you made a nice wave slider!) would need a "close" button.

**Comment by Feichtmeier** [comment]
> After ~10/15 navigations between random pages, the UI starts to be laggy. It could be wise to use replace instead of push. I suspect the "Album Grid page" to be heavy. It has to keep in memory dozens of images that are more or less heavy. 

I can not verify this for me, either on a super slow intel from years ago and also not on my older 7a pixel phone I can navigate as long as I want and I dont get any stutter / lag

> If the track has no album cover, it shows an icon/widget different than the one from the Album page. It's black isntead of being green.

well musicpod needs your audio files to be propertly tagged :P but didnt the fix for the picture in the directory help? you mean when some track of some album has a cover, there is no cover in the directory and then some other track from the same album has no cover?

> The favorites should be easy to access. It should be the first one.

No, I disagree. MusicPod is not only a local audio player and I prioritize all 3 features (local, radio and podcasts) equally

> Maybe some Hero animation could be nice. For example, if you press the album cover from the listening bar, the image move to the new position.

Yes this is on my todolist but I dont have an issue for this. If you like you could create an issue for this?

> I don't think the "back arrow" in the AppBar is useful. I already have a back button that does the same thing and it's easier to reach.

I also disagree here. This is dedicated to navigate always back from everywhere, this is the same on desktop and I love this. WOnt change :)

**Comment by ClementBeal** [comment]
> > After ~10/15 navigations between random pages, the UI starts to be laggy. It could be wise to use replace instead of push. I suspect the "Album Grid page" to be heavy. It has to keep in memory dozens of images that are more or less heavy.
> 
> I can not verify this for me, either on a super slow intel from years ago and also not on my older 7a pixel phone I can navigate as long as I want and I dont get any stutter / lag

It seems it's not slowing down if I close the back with the back button. If I act as a the stupid QA tester and press the BottomNavigationBar several times on the albums pages without closing the pages, it will be slower. It's actually a very special situation that shouldn't be occur (excepting if people are navigating a lot between local/radio/podcast) 

> > If the track has no album cover, it shows an icon/widget different than the one from the Album page. It's black isntead of being green.
> 
> well musicpod needs your audio files to be propertly tagged :P but didnt the fix for the picture in the directory help? you mean when some track of some album has a cover, there is no cover in the directory and then some other track from the same album has no cover?

Sometimes, the music cover can be missing even if the album folder. Sorry, I just saw that my sentence wasn't right lol. Here are 2 images to show you what I'm talking about :
![1](https://github.com/user-attachments/assets/5847ccff-2de5-45d6-9f55-cdb30acfcd10)
![2](https://github.com/user-attachments/assets/bc8e22db-c816-43b2-b67d-05e7e08be25c)

The missing cover icon/widget is not the same.

> > The favorites should be easy to access. It should be the first one.
> 
> No, I disagree. MusicPod is not only a local audio player and I prioritize all 3 features (local, radio and podcasts) equally

Sorry, I mean the first one in the Local filter selector. 
From ` Titles | Artists | Albums |...` to `Favorites | Titles | Artists | ...`

> > Maybe some Hero animation could be nice. For example, if you press the album cover from the listening bar, the image move to the new position.
> 
> Yes this is on my todolist but I dont have an issue for this. If you like you could create an issue for this?

Yep let me do it

> > I don't think the "back arrow" in the AppBar is useful. I already have a back button that does the same thing and it's easier to reach.
> 
> I also disagree here. This is dedicated to navigate always back from everywhere, this is the same on desktop and I love this. WOnt change :)
